File: py/prob_3.py
class Solution:
    def lengthOfLongestSubstring(self, s):
        """
        :type s: str
        :rtype: int
        """
        
        n = len(s)
        ans = 0
        index = [0] * 128
        
        i = 0
        for j in range(n):
            i = max(index[ord(s[j])], i)
            ans = max(ans, j - i + 1)
            index[ord(s[j])] = j + 1
        return ans

File: cpp/prob_3.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int n = s.length();
        vector<int> chartable(128, 0);
        int ans = 0;
        
        for (int i=0, j=0; j < n; j++) {
            i = max(chartable[int(s[j])], i);
            ans = max(ans, j-i+1);
            chartable[int(s[j])] = j+1;
        };
        
        return ans;  
    };
};
